### PR TITLE
Add the Zeroday template

### PR DESCRIPTION
**Pull request by zeeket**
This PR adds the Zeroday templat. The template creates a sample fullstack web application using node.js and React (among other technologies) which can be deployed to a managed Kubernetes cluster hosted by Amazon.

More information and documentation is provided in the template README files.

### CODE REVIEW

File: applications/web_app/fullstack/zeroday-fullstack/template/backend/test/auth.test.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    await agent.post('/api/reset_password').send({ token: undefined, newPassword: 'monkeyBarrel2000!', verifyPassword: 'monkeyBarrel2000!' }).expect(401);
+  });
+});
+
+/*
+describe('Facebook Login', () => {
+  var testUsers = {
+    '106703763266783': {
+      name: 'Dave Alaghadgicfee Valtchanovwitz',
+      email: 'rmknpulqlu_1496817297@tfbnw.net',
```
Comment (romuaa) [inline applications/web_app/fullstack/zeroday-fullstack/template/backend/test/auth.test.js:80]: Also here @.invalid

File: applications/web_app/fullstack/zeroday-fullstack/template/backend/test/factories.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const _ = require('lodash');
+const database = require('../src/database');
+
+let i = 0;
+
+const email = () => {
+  i += 1;
+  return `test${i}@test.com`;
```
Comment (romuaa) [inline applications/web_app/fullstack/zeroday-fullstack/template/backend/test/factories.js:8]: Also here @.invalid

File: applications/web_app/fullstack/zeroday-fullstack/template/frontend/package.json.j2 [hunk 3]
```diff
@@ -0,0 +1,67 @@
+{
+  "name": "{{ctx.current_template['repository_name']}}_frontend",
+  "version": "0.0.1",
+  "description": "Node React Project Template",
+  "author": "Marko Klemetti <marko.klemetti@eficode.com>",
```
Comment (romuaa) [inline applications/web_app/fullstack/zeroday-fullstack/template/frontend/package.json.j2:5]: Also Here Eficode ROOT, Eficode, Sprout r something else than our CTO?

File: applications/web_app/fullstack/zeroday-fullstack/template/robot/resources/common.robot [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  Submit Form
+
+Logout
+  Run Keyword And Ignore Error    Go To    ${SERVER}/logout
+
+Click Button With Span  [Arguments]  ${locator}
+  Click Button  //button[contains(text(), '${locator}')]
+
+Verify language is suomi
+  Wait Until Page Contains    Salasana
```
Comment (romuaa) [inline applications/web_app/fullstack/zeroday-fullstack/template/robot/resources/common.robot:75]: So the UI is multilanguage? fi, se, en? If so I think it is fine.

File: applications/web_app/fullstack/zeroday-fullstack/template/robot/resources/variables.robot [hunk 5]
```diff
@@ -0,0 +1,12 @@
+*** Variables ***
+
+${VALID EMAIL}            jalmari.pipo@eficode.com
```
Comment (romuaa) [inline applications/web_app/fullstack/zeroday-fullstack/template/robot/resources/variables.robot:3]: Replace with invalid domains e.g. @xxx.invalid

File: applications/web_app/fullstack/zeroday-fullstack/template/robot/resources/variables.robot [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+${VALID PASSWORD}         test1234
+
+${REGISTER EMAIL}         siljamari.pipo@eficode.com
```
Comment (romuaa) [inline applications/web_app/fullstack/zeroday-fullstack/template/robot/resources/variables.robot:6]: Replace with invalid domains e.g. @xxx.invalid

File: applications/web_app/fullstack/zeroday-fullstack/template/frontend/src/components/navigation/NavigationMain.jsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+const handleLogout = (dispatch) => dispatch(logout());
+
+const NavigationMain = (props) => {
+  const { user, history, dispatch } = props;
+
+  return (
+    <Menu borderless fixed="top">
+      <Menu.Item className="header" onClick={() => handleHome(history)}>
+        <span className="logo" />
+        <img alt="" className="navbar-logo" src="images/eficode_logo_black_200x150.png" />
```
Comment (romuaa) [inline applications/web_app/fullstack/zeroday-fullstack/template/frontend/src/components/navigation/NavigationMain.jsx:23]: @mikkilevon @venetastefanova What about template branding? Do we want Eficode logos on the template example UI?

**Review by romuaa — CHANGES_REQUESTED** [review]

File: applications/web_app/fullstack/zeroday-fullstack/template/backend/package.json.j2 [hunk 8]
```diff
@@ -0,0 +1,65 @@
+{
+  "name": "{{ctx.current_template['repository_name']}}_backend",
+  "version": "0.0.1",
+  "description": "Node React Project Template",
+  "author": "Jan Tuomi <jan.tuomi@eficode.com>, Marko Klemetti <marko.klemetti@eficode.com>",
```
[the remark this answers is not in the document]
Comment (jpvg10) [reply applications/web_app/fullstack/zeroday-fullstack/template/backend/package.json.j2:5]: Which one should I put? Is there a discussion to be had with someone to decide this?

File: applications/web_app/fullstack/zeroday-fullstack/template/backend/package.json.j2 [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  "license": "UNLICENSED",
```
[the remark this answers is not in the document]
Comment (jpvg10) [reply applications/web_app/fullstack/zeroday-fullstack/template/backend/package.json.j2:6]: Also here, is there a discussion to be had with someone?

File: applications/web_app/fullstack/zeroday-fullstack/template/robot/resources/common.robot (hunk 4 above)
Comment (jpvg10) [reply applications/web_app/fullstack/zeroday-fullstack/template/robot/resources/common.robot:75]: It is. The login page has a dropdown to switch between all 3

File: applications/web_app/fullstack/zeroday-fullstack/template/backend/db/seeds.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
+const db = require('../src/database');
+const logger = require('../src/logger');
+
+const create = async () => {
+  try {
+    await db.sync({ force: true });
+    await db.User.create({
+      firstname: 'Jalmari', lastname: 'Pipo', email: 'jalmari.pipo@eficode.com', password: 'test1234',
```
[the remark this answers is not in the document]
Comment (jpvg10) [reply applications/web_app/fullstack/zeroday-fullstack/template/backend/db/seeds.js]: Done

File: applications/web_app/fullstack/zeroday-fullstack/template/backend/test/auth.test.js (hunk 1 above)
Comment (jpvg10) [reply applications/web_app/fullstack/zeroday-fullstack/template/backend/test/auth.test.js]: Done

File: applications/web_app/fullstack/zeroday-fullstack/template/backend/test/factories.js (hunk 2 above)
Comment (jpvg10) [reply applications/web_app/fullstack/zeroday-fullstack/template/backend/test/factories.js]: Done

File: applications/web_app/fullstack/zeroday-fullstack/template/robot/resources/variables.robot (hunk 5 above)
Comment (jpvg10) [reply applications/web_app/fullstack/zeroday-fullstack/template/robot/resources/variables.robot]: Done

File: applications/web_app/fullstack/zeroday-fullstack/template/robot/resources/variables.robot (hunk 6 above)
Comment (jpvg10) [reply applications/web_app/fullstack/zeroday-fullstack/template/robot/resources/variables.robot]: Done

File: applications/web_app/fullstack/zeroday-fullstack/template.json [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+            "volumes": [
+                "./frontend/src:/usr/src/app/src"
+            ],
+            "ports": [
+                "80:8000"
+            ]
+        }
+    },
+    "postActions": {
+        "Jenkins": ["To set up the Jenkins pipeline, run the 'Jenkins' executor script in Sprout."]
```
[the remark this answers is not in the document]
Comment (mikkilevon) [reply applications/web_app/fullstack/zeroday-fullstack/template.json:75]: Yeah.

[the remark this answers is not in the document]
Comment (zeeket) [reply applications/web_app/fullstack/zeroday-fullstack/template.json]: done 👍

File: applications/web_app/fullstack/zeroday-fullstack/template/frontend/src/components/navigation/NavigationMain.jsx (hunk 7 above)
Comment (romuaa) [reply applications/web_app/fullstack/zeroday-fullstack/template/frontend/src/components/navigation/NavigationMain.jsx:23]: So the Eficode stuff should be replaced with some ACME CORP Logos?
@mikkilevon @zeeket @jpvg10

Comment (zeeket) [reply applications/web_app/fullstack/zeroday-fullstack/template/frontend/src/components/navigation/NavigationMain.jsx:23]: If the customer is going to replace it anyways, does it matter if we use ACME CORP or Eficode as the placeholder? It seems like most "generic company logo" material on the internet has a commercial use license that costs a little bit of money. Ideally the placeholder would look good enough to be presentable in demos.

**Review by romuaa — APPROVED** [review]

**Review by mikkilevon — APPROVED** [review]

**Pull request merged**